src/nvim/api/vim.c

```diff
@@ -74,6 +74,72 @@ void api_vim_free_all_mem(void)
   map_free(String, handle_T)(namespace_ids);
 }
 
+/// Posts a message to the global logger (which by default synchronously writes
+/// to |$NVIM_LOG_FILE| using a best-effort, naive implementation).
+///
+/// Message is ignored (not logged) if `level` is not "error", unless 'verbose'
+/// is set.
+///
+/// Example log message:
+///
+/// <pre>
+///   ERROR 2020-01-12T01:56:19.484 93296 pynvim:remote: connected…
+///   ^level ^timestamp             ^pid  ^category      ^message
+/// </pre>
+///
+/// @see |nvim_get_chan_info()|
+/// @see |nvim_set_client_info()|
+///
+/// @param channel_id Channel id (implicit dispatcher arg)
+/// @param level  Log level (from highest to lowest: "error", "warn", "info",
+///   "debug").  If 'verbose' is not set, only "error" messages are written to
+///   |$NVIM_LOG_FILE|.
+/// @param msg  Message body
+/// @param opt  Map of optional parameters:
+///   - type: String name of the subsystem or grouping (e.g. "UI", "RPC",
+///     "MyPlugin", …). For API clients this defaults to the `client.name`
+///     field of |nvim_get_chan_info()|.
+///   - join: (boolean, default=false)
+///   - trunc: (integer, default=0) Constrain the message body to this size.
+void nvim_log(uint64_t channel_id, String level, String msg, Dictionary opt,
+              Error *err)
+  FUNC_API_SINCE(7)
+{
+  int loglevel = log_level_from_name(level.data);
+  if (loglevel == -1) {
+    api_set_error(err, kErrorTypeValidation, "invalid level: %s", level.data);
+    return;
+  }
+  Object rv = OBJECT_INIT;
+  Array args = ARRAY_DICT_INIT;
+  ADD(args, DICTIONARY_OBJ(opt));
+  rv = nvim_exec_lua(STATIC_CSTR_AS_STRING(
+      "local o = ...\n"
+      "vim.validate{trunc={o.trunc,'n',true},join={o.join,'b',true}}\n"
+      "return {(o.join and true or false), (o.trunc and o.trunc or 0),\n"
+      "  (o.type and o.type or '')}\n"),
+                     args, err);
+  if (ERROR_SET(err)) {
+    goto theend;
+  }
+  assert(rv.type == kObjectTypeArray);
+  Object *items = rv.data.array.items;
+  bool join = items[0].data.boolean;
+  int64_t trunc = items[1].data.integer >= 0 ? items[1].data.integer : 0;
+  char *prefix = items[2].data.string.data;
+
+  char p[32];  // Formatted prefix.
+  if (prefix[0] == '\0') {
+    // Generate a default category/group/prefix.
+    rpc_chan_desc(channel_id, prefix, sizeof(p) - 2);
```

Comment:
> If someone could help with the travis build issue, the PR is pretty much complete.

Perhaps this is where the problem lies. I don't think the length of the prefix is 32 bytes. I haven't looked closely, but how about passing `p` to `rpc_chan_desc` as shown below?

```C
  if (prefix[0] == '\0') {
    // Generate a default category/group/prefix.
    rpc_chan_desc(channel_id, p, sizeof(p) - 2);
  } else {
    snprintf(p, sizeof(p), "%s: ", prefix);
  }
```

Reply:
Sorry for the late response :sweat_smile: 
I didn't get a notification from github regarding this and only came across this today when I came here to try and check things again. I will try this and run the tests again.

Reply:
Thanks a lot @erw7. That fixed it. Now there is this memory leak to debug.
Thank you once again. I was stuck on this from more than a week, and never even looked at that line.I just saw the first few errors and thought they were to blame :/ 
Astute observation I must say :)